### PR TITLE
feature/OE-231-setter-comission-doesnt-showing-on-average-results

### PR DESCRIPTION
**Pull request by devsquad-mathews-lima**
## Review

For the submitter, initial self-review:

- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency between the changeset and the goal stated above
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Complexity - is the proposed change introducing any unjustified complexity? 
- [ ] Thoroughness of added tests and any missing edge cases

### CODE REVIEW

File: app/Http/Livewire/Reports/ReportsOverview.php [hunk 1]
```diff
@@ -186,20 +185,36 @@ public function getAvgSystemSize(Collection $customers)
...
-        return $customers->sum(function ($customer) {
-            return $this->getSetterCommission($customer);
-        });
+        return $customers
+            ->when(user()->hasRole('Setter'), function (Collection $customers) {
+                return $customers->where('setter_id', user()->id);
+            })
+            ->when(user()->notHaveRoles(['Setter', 'Admin', 'Owner']), function (Collection $customers) {
+                return $customers->where('sales_rep_id', user()->id);
+            })
```
Comment (Ron-Junior) [inline app/Http/Livewire/Reports/ReportsOverview.php:194]: when the user have the other roles the sum still will be all customers that the user is a setter

File: app/Http/Livewire/Reports/ReportsOverview.php [hunk 2]
```diff
@@ -186,20 +185,36 @@ public function getAvgSystemSize(Collection $customers)
...
-        return $customers->avg(function ($customer) {
-            return $this->getSetterCommission($customer);
-        });
+        return $customers
+            ->when(user()->hasRole('Setter'), function (Collection $customers) {
+                return $customers->where('setter_id', user()->id);
+            })
+            ->when(user()->notHaveRoles(['Setter', 'Admin', 'Owner']), function (Collection $customers) {
+                return $customers->where('sales_rep_id', user()->id);
+            })
```
Comment (Ron-Junior) [inline app/Http/Livewire/Reports/ReportsOverview.php:211]: When the user has the other roles the average still will be all customers that the user is a setter.

**Review by Ron-Junior — CHANGES_REQUESTED** [review]

File: app/Http/Livewire/Reports/ReportsOverview.php (hunk 2 above)
Comment (Ron-Junior) [reply app/Http/Livewire/Reports/ReportsOverview.php:211]: I'm thinking. maybe this problem too occurs with the sales rep commission

**Review by Ron-Junior — APPROVED** [review]
